src/gc/gc.cpp

```diff
@@ -33623,6 +33620,8 @@ HRESULT GCHeap::Initialize ()
     }
 
     g_gc_pFreeObjectMethodTable = GCToEEInterface::GetFreeObjectMethodTable();
+    g_num_processors = GCToOSInterface::GetCurrentProcessCpuCount();
```

Comment:
GetLogicalCpuCount is for something different - that's for getting how many logical CPUs are on one physical processor. You could just add a new API for getting the # of processors on the machine.